diff --git a/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py b/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py
index d50599e..b0e8794 100644
--- a/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py
+++ b/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py
@@ -1429,10 +1429,7 @@ class nnUNetTrainer(object):
 
         ### ADAPTED FOR KAAPANAFED ###
         if os.getenv("PREP_INCREMENT_STEP", None) == "from_dataset_properties":
-            self.save_checkpoint(join(self.output_folder, "model_final_checkpoint.model"))
-            # nnunet_trainer.save_checkpoint(
-            #     join(nnunet_trainer.output_folder, "model_final_checkpoint.model")
-            # )
+            self.save_checkpoint(join(self.output_folder, "checkpoint_final.pth"))
             print("Preparation round done: Aggregation of model initialization!")
             return
         ##############################
-- 