diff --git a/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py b/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py
index b0e8794..767ef66 100644
--- a/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py
+++ b/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py
@@ -1219,7 +1219,8 @@ class nnUNetTrainer(object):
 
     def save_checkpoint(self, filename: str) -> None:
         if self.local_rank == 0:
-            if not self.disable_checkpointing:
+            # self.disable_checkpointing disables intermediate checkpointing but not final checkpointing
+            if not self.disable_checkpointing or "checkpoint_final.pth" in filename:
                 if self.is_ddp:
                     mod = self.network.module
                 else:
-- 