diff --git a/nnunetv2/run/run_training.py b/nnunetv2/run/run_training.py
index 3612711..af5aaea 100644
--- a/nnunetv2/run/run_training.py
+++ b/nnunetv2/run/run_training.py
@@ -206,9 +206,13 @@ def run_training(dataset_name_or_id: Union[str, int],
         if not only_run_validation:
             nnunet_trainer.run_training()
 
-        if val_with_best:
-            nnunet_trainer.load_checkpoint(join(nnunet_trainer.output_folder, 'checkpoint_best.pth'))
-        nnunet_trainer.perform_actual_validation(export_validation_probabilities)
+        if fold == "all":
+            print("--> fold == 'all'")
+            print("--> DONE")
+        else:
+            if val_with_best:
+                nnunet_trainer.load_checkpoint(join(nnunet_trainer.output_folder, 'checkpoint_best.pth'))
+            nnunet_trainer.perform_actual_validation(export_validation_probabilities)
 
 
 def run_training_entry():
